storage/storage_test.go

```diff
@@ -10,98 +10,102 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-func makeTestStorage(t *testing.T, initDir string) Storage {
-	err := os.RemoveAll("../_bucketdata")
+const tempDir = "../_bucketdata"
+
+func makeTestStorage(t *testing.T, initDir string) Bucket {
+	err := os.RemoveAll(tempDir)
 	require.NoError(t, err)
-	err = os.Mkdir("../_bucketdata", 0755)
+	err = os.Mkdir(tempDir, 0755)
 	require.NoError(t, err)
 
-	bucket, err := pail.NewLocalBucket(pail.LocalOptions{
-		Path:   "../_bucketdata",
-		Prefix: "",
+	bucket, err := NewBucket(BucketOpts{
+		Location: PailLocal,
+		Path:     tempDir,
 	})
 	require.NoError(t, err)
 
-	err = bucket.Push(context.Background(), pail.SyncOptions{
-		Local:  initDir,
-		Remote: "/",
-	})
-	require.NoError(t, err)
+	if initDir != "" {
+		err = bucket.Push(context.Background(), pail.SyncOptions{
+			Local:  initDir,
+			Remote: "/",
+		})
+		require.NoError(t, err)
+	}
 
-	return NewStorage(bucket)
+	return bucket
 }
 
 func TestBasicStorage(t *testing.T) {
 	storage := makeTestStorage(t, "../testdata/simple")
-	defer os.RemoveAll("../_bucketdata")
-	results, err := storage.bucket.Get(context.Background(), "5a75f537726934e4b62833ab6d5dca41/metadata.json")
+	defer os.RemoveAll(tempDir)
+	results, err := storage.Get(context.Background(), "5a75f537726934e4b62833ab6d5dca41/metadata.json")
 	assert.NoError(t, err)
 	assert.NotEqual(t, nil, results)
 
 }
 
-func TestGetTestLogLines(t *testing.T) {
-	storage := makeTestStorage(t, "../testdata/simple")
-	defer os.RemoveAll("../_bucketdata")
-	iterator, err := storage.GetTestLogLines(context.Background(), "5a75f537726934e4b62833ab6d5dca41", "62dba0159041307f697e6ccc")
-	require.NoError(t, err)
+func TestGetS3Options(t *testing.T) {
```

Comment:
If we want to use S3 for our tests then we are definitely going to need to add a "cleanup" method that knows how to handle that.